src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,46 @@
-'use strict';
+('use strict');
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  // eslint-disable-next-line no-useless-constructor
+  constructor(name) {
+    super(name);
+  }
+
+  bite(herbivore) {
+    if (!herbivore.hidden && !(herbivore instanceof Carnivore)) {
```

Comment:
The condition inside the `if` statement should only check if the `herbivore` is an instance of `Herbivore` and not hidden. The current condition also checks if the `herbivore` is not an instance of `Carnivore`, which is unnecessary according to the task requirements.